src/main/java/ch/njol/skript/util/Timespan.java

```diff
@@ -32,49 +34,84 @@
 import ch.njol.util.coll.CollectionUtils;
 import ch.njol.yggdrasil.YggdrasilSerializable;
 
-/**
- * @author Peter Güttinger
- * @edited by Mirreducki. Increased maximum timespan.
- */
 public class Timespan implements YggdrasilSerializable, Comparable<Timespan> { // REMIND unit
 
-	private final static Noun m_tick = new Noun("time.tick");
-	private final static Noun m_second = new Noun("time.second");
-	private final static Noun m_minute = new Noun("time.minute");
-	private final static Noun m_hour = new Noun("time.hour");
-	private final static Noun m_day = new Noun("time.day");
-	private final static Noun m_week = new Noun("time.week");
-	private final static Noun m_month = new Noun("time.month");
-	private final static Noun m_year = new Noun("time.year");
-	final static Noun[] names = {m_tick, m_second, m_minute, m_hour, m_day, m_week, m_month, m_year};
-	final static long[] times = {50L, 1000L, 1000L * 60L, 1000L * 60L * 60L, 1000L * 60L * 60L * 24L,  1000L * 60L * 60L * 24L * 7L,  1000L * 60L * 60L * 24L * 30L,  1000L * 60L * 60L * 24L * 365L};
-	final static HashMap<String, Long> parseValues = new HashMap<>();
+
+	private enum Times {
+
+		TICK("time.tick", 50L),
+		SECOND("time.second", 1000L),
+		MINUTE("time.minute", SECOND.time * 60L),
+		HOUR("time.hour", MINUTE.time * 60L),
+		DAY("time.day", HOUR.time * 24L),
+		WEEK("time.week", DAY.time * 7L),
+		MONTH("time.month", DAY.time * 30L),
+		YEAR("time.year", DAY.time * 365L);
+
+		private final Noun name;
+		private final long time;
+
+		Times(String name, long time) {
+			this.name = new Noun(name);
+			this.time = time;
+		}
+
+	}
+
+	@SuppressWarnings("unchecked")
+	static final NonNullPair<Noun, Long>[] simpleValues = new NonNullPair[] {
+		new NonNullPair<>(Times.YEAR.name, Times.YEAR.time),
+		new NonNullPair<>(Times.MONTH.name, Times.MONTH.time),
+		new NonNullPair<>(Times.WEEK.name, Times.WEEK.time),
+		new NonNullPair<>(Times.DAY.name, Times.DAY.time),
+		new NonNullPair<>(Times.HOUR.name, Times.HOUR.time),
+		new NonNullPair<>(Times.MINUTE.name, Times.MINUTE.time),
+		new NonNullPair<>(Times.SECOND.name, Times.SECOND.time)
+	};
```

Comment:
```suggestion
	static final List<NonNullPair<Noun, Long>> simpleValues = Arrays.asList(new NonNullPair[] {
		new NonNullPair<>(Times.YEAR.name, Times.YEAR.time),
		new NonNullPair<>(Times.MONTH.name, Times.MONTH.time),
		new NonNullPair<>(Times.WEEK.name, Times.WEEK.time),
		new NonNullPair<>(Times.DAY.name, Times.DAY.time),
		new NonNullPair<>(Times.HOUR.name, Times.HOUR.time),
		new NonNullPair<>(Times.MINUTE.name, Times.MINUTE.time),
		new NonNullPair<>(Times.SECOND.name, Times.SECOND.time)
	});
```
We don't want modification later on.